### ISSUE TITLE
Calling godep inside of a pre-commit hook fails

### ISSUE THREAD

**Issue by kouno** [tools/godep#190]
In my `.git/hooks/pre-commit` file I have the following:

```bash
#!/bin/bash

set -e

cd "$(godep path)/../.." && godep save -r ./...
godep_imports="$(git diff | grep 'Godeps/_workspace' | wc -l)"

if [ "$godep_imports" -gt 0 ]; then
  cat <<\EOF
Error: Some Go dependencies have been updated with `godep save -r ./...`

Please add them to your commit.
EOF
  exit 1
fi

exit 0
```

It fails with:
```
$ git commit
# cd /Users/kouno/.gocode/src/github.com/flynn/flynn/pkg/random; git rev-parse HEAD
fatal: Not a git repository: '.git'
godep: exit status 128
godep: error loading dependencies
```

On my terminal, running the same command manually works:
```
$ cd /Users/kouno/.gocode/src/github.com/flynn/flynn/pkg/random; git rev-parse HEAD
736b4d597fe39e75c9777942245ee02a24bb0d26
```

To be fair it looks like it's a `git` issue, because when I run:
```bash
#!/bin/bash

set -e

cd /Users/kouno/.gocode/src/github.com/flynn/flynn/pkg/random
git rev-parse HEAD

exit 0
```

I get:
```
$ git commit
fatal: Not a git repository: '.git'
```

And this works fine if I run `git rev-parse HEAD` from the root folder instead (i.e. `$GOPATH/src/github.com/flynn/flynn/`). I couldn't find a workaround to this (maybe setting GIT_DIR before calling `git rev-parse` would help?).

**Comment by kouno** [comment]
And obviously I found the answer just after posting this...

I just had to unset GIT_DIR and everything works fine:
```bash
#!/bin/bash

set -e

unset GIT_DIR

cd "$(godep path)/../.."
godep save -r ./...
godep_imports="$(git diff | grep 'Godeps/_workspace' | wc -l)"

if [ "$godep_imports" -gt 0 ]; then
  cat <<\EOF
ERROR: Some Go dependencies have been updated with `godep save -r ./...`
ERROR: Please add them to your commit.
EOF
  exit 1
fi

exit 0
```

Sorry for the noise. :disappointed:

**Comment by ErikDubbelboer** [comment]
Would be better if this gets fixed with #169
